lib/auth/user.go

```diff
@@ -24,38 +24,47 @@ limitations under the License.
 package auth
 
 import (
-	"github.com/gravitational/teleport"
+	"context"
+
 	"github.com/gravitational/teleport/lib/events"
 	"github.com/gravitational/teleport/lib/services"
 
 	"github.com/gravitational/trace"
 )
 
-// UpsertUser upserts user
+// CreateUser inserts a new user entry in a backend.
+func (s *AuthServer) CreateUser(ctx context.Context, user services.User) error {
+	if err := s.Identity.CreateUser(user); err != nil {
+		return trace.Wrap(err)
+	}
+
+	s.EmitAuditEvent(events.UserCreate, events.EventFields{
+		events.EventUser:       user.GetCreatedBy().User.Name,
+		events.UserExpires:     user.Expiry(),
+		events.UserRoles:       user.GetRoles(),
+		events.UserCreatedName: user.GetName(),
+	})
+
+	return nil
+}
+
+// UpsertUser updates a user.
 func (s *AuthServer) UpsertUser(user services.User) error {
 	err := s.Identity.UpsertUser(user)
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
-	// If the user was successfully upserted, emit an event.
-	var connectorName string
-	if user.GetCreatedBy().Connector == nil {
-		connectorName = teleport.Local
-	} else {
-		connectorName = user.GetCreatedBy().Connector.ID
-	}
 	s.EmitAuditEvent(events.UserUpdate, events.EventFields{
-		events.EventUser:     user.GetName(),
-		events.UserExpires:   user.Expiry(),
-		events.UserRoles:     user.GetRoles(),
-		events.UserConnector: connectorName,
+		events.EventUser:   user.GetName(),
```

Comment:
It looks like we do need to include connector ID as well (teleport calls these method for SSO users)

```golang
	var connectorName string	
	if user.GetCreatedBy().Connector == nil {	
		connectorName = teleport.Local	
	} else {	
		connectorName = user.GetCreatedBy().Connector.ID	
	}
```